src/main/scala/com/softwaremill/realworld/articles/ArticlesRepository.scala

```diff
@@ -1,38 +1,48 @@
 package com.softwaremill.realworld.articles
 
-import zio.{UIO, ZIO, ZLayer}
+import com.softwaremill.realworld.profiles.ProfileRow
+import io.getquill.*
+import zio.{IO, UIO, ZIO, ZLayer}
 
+import java.sql.SQLException
 import java.time.Instant
+import javax.sql.DataSource
 
-class ArticlesRepository:
+class ArticlesRepository(quill: SqliteZioJdbcContext[SnakeCase]):
 
-  // TODO user proper db or create in-memory thread-safe store
-  private val articles: List[StoredArticle] = List(
-    StoredArticle(
-      1,
-      "dummy-article",
-      "Dummy Title",
-      "Represents a dummy article.",
-      "Dummy body",
-      List("dummy", "article"),
-      Instant.now(),
-      Instant.now(),
-      10
-    ),
-    StoredArticle(
-      2,
-      "dummy-article-2",
-      "Dummy Title 2",
-      "Represents a dummy article 2.",
-      "Dummy body 2",
-      List("dummy", "article", "2"),
-      Instant.now(),
-      Instant.now(),
-      10
-    )
+  import quill._
+  def list(): ZIO[DataSource, SQLException, List[Article]] = run(
+    for {
+      ar <- querySchema[ArticleRow](entity = "articles")
+      pr <- querySchema[ProfileRow](entity = "users") if pr.userId == ar.authorId
```

Comment:
why `ProfileRow` and not `UserRow`? Is `profile` used as a concept elsewhere in the domain? So far there have been `user`s and `author`s :)

Reply:
Yes `Profile` can be author and a follower, it is a user from the point of view of another user: https://realworld-docs.netlify.app/docs/specs/backend-specs/api-response-format#profile
User is more like an account that can log in/out: https://realworld-docs.netlify.app/docs/specs/backend-specs/api-response-format#users-for-authentication

Reply:
Ok, if it's a domain concept, that's fine then